Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/connectmgr/persistance/dao/ConnectionManagerDAOBase.java

```diff
@@ -56,36 +61,44 @@
+    protected BusinessDetail loadBusinessDetail(File file) throws JAXBException {
         BusinessDetail resp = null;
         try {
-
             synchronized (file) {
 
-                final SAXParserFactory spf = SAXParserFactory.newInstance();
-                spf.setNamespaceAware(true);
-                spf.setFeature(NhincConstants.FEATURE_GENERAL_ENTITIES, false);
-                spf.setFeature(NhincConstants.FEATURE_DISALLOW_DOCTYPE, true);
-                spf.setFeature(NhincConstants.FEATURE_PARAMETER_ENTITIES, false);
-
-                final XMLReader xmlReader = spf.newSAXParser().getXMLReader();
-                final InputSource inputSource = new InputSource(new FileReader(file));
-                final SAXSource source = new SAXSource(xmlReader, inputSource);
-
-                final JAXBContext context = JAXBContext.newInstance(BusinessDetail.class);
-                final Unmarshaller unmarshaller = context.createUnmarshaller();
-
-                final InputSource stream = source.getInputSource();
-
-                final Reader reader = stream.getCharacterStream();
-
-                final JAXBElement<BusinessDetail> jaxbElement = unmarshaller.unmarshal(new StreamSource(reader),
+                JAXBContext context = JAXBContext.newInstance(BusinessDetail.class);
+                final DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+
+                dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+                dbf.setFeature(NhincConstants.FEATURE_GENERAL_ENTITIES, false);
+                dbf.setFeature(NhincConstants.FEATURE_DISALLOW_DOCTYPE, true);
+                dbf.setFeature(NhincConstants.FEATURE_PARAMETER_ENTITIES, false);
+                dbf.setExpandEntityReferences(false);
+
+                // Convert File to Document
+                final DocumentBuilder db = dbf.newDocumentBuilder();
+                final Document doc = db.parse(file);
+                doc.getDocumentElement().normalize();
+                DOMSource source = new DOMSource(doc);
+
+                // Convert Document to StreamSource
+                TransformerFactory factory = TransformerFactory.newInstance();
+                Transformer transformer = factory.newTransformer();
+                StreamResult result = new StreamResult();
+                ByteArrayOutputStream writer = new ByteArrayOutputStream();
+                result.setOutputStream(writer);
+                transformer.transform(source, result);
+                ByteArrayInputStream reader = new ByteArrayInputStream(writer.toByteArray());
+
+                // Unmarshal
+                Unmarshaller unmarshaller = context.createUnmarshaller();
+                JAXBElement<BusinessDetail> jaxbElement = unmarshaller.unmarshal(new StreamSource(reader),
                         BusinessDetail.class);
-
                 resp = jaxbElement.getValue();
-
+                reader.close();
```

Comment:
what happen if exception happen before close statement?  recommend to move into final block